### DISCUSSION TITLE
Is there a way to bind css class to a variable or condition?

### DISCUSSION THREAD

**Discussion by ghost** [zauberzeug/nicegui#2644] [Q&A] [answered]
### Question

Hi, everybody!

I would like to know whether there is a way to bind a css class of a ui element to a definite variables value. For example... I have a menu with a couple of items . Besides I have a variable "current_menu_item" where I store the name of active menu item. Every time I click on menu item the background color have to change (adding css class bg-gray-100). And the bg color becomes gray only if the name ot the clicked item and the variable value are equal.

As I know there is a v-bind in VueJS. Using it I can bind a css class to a variable and change it conditionally. So is there an alternative to it in NiceGUI?

**Comment by falkoschindler** [comment]
Hi @zagirovaa,

That's a bit tricky. You could bind to the private `_style` dictionary, but need to call the `update` method after changing a value in this dictionary:
```py
label = ui.label('Hi!')
ui.checkbox('red', on_change=label.update) \
    .bind_value_to(label._style, 'background-color', lambda x: 'red' if x else 'white')
```

A simple event handler might be easier:
```py
label = ui.label('Hi!')
ui.checkbox('red', on_change=lambda e: label.style(f'background-color: {"red" if e.value else "white"}'))
```

You can also use Tailwind or Quasar classes:
```py
label = ui.label('Hi!')
ui.checkbox('red', on_change=lambda e: label.classes(add='bg-red' if e.value else None,
                                                     remove=None if e.value else 'bg-red'))
```

**Reply by ghost** [reply]
Hi!

As I think the second and the third variants won't work in my case cause except for making background of an active item gray, I need to remove gray background from the previous active one.

I will give a try with the first example using private _style dictionary.

**Reply by ghost** [reply]
Hi @falkoschindler,

Unfortunately the first example doesn't work either.

**Reply by ghost** [reply, marked as answer]
Hi @falkoschindler ,

Finally I found a solution.

I moved all the code related to creating menu with its items to a separate function and added @ui.refreshable decorator to it. Each time the menu item is clicked the whole menu is recreated (refreshed). Now everything works like a charm!
